### PR TITLE
feat(wasm): support getting request body

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by tzssangglass — DISMISSED** [review]

**Review by bzp2010 — APPROVED** [review]

File: t/APISIX.pm [hunk 1]
```diff
@@ -217,6 +217,13 @@ if ($version =~ m/\/apisix-nginx-module/) {
 _EOC_
 }
 
+my $a6_ngx_vars = "";
+if ($version =~ m/\/apisix-nginx-module/) {
+    $a6_ngx_vars = <<_EOC_;
```
Comment (leslie-tsang) [inline t/APISIX.pm:222]:
```suggestion
    $a6_ngx_vars .= <<_EOC_;
```
Would this be better ? If there is a new setting that needs to be applied, we don't need to change it.

File: t/wasm/request-body.t [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+
+
+=== TEST 5: hit
+--- request
+POST /hello
+hello
+--- grep_error_log eval
+qr/request get body: \w+/
+--- grep_error_log_out
```
Comment (leslie-tsang) [inline t/wasm/request-body.t:152]: Is it possible to remove this ?

File: t/APISIX.pm [hunk 3]
```diff
@@ -217,6 +217,13 @@ if ($version =~ m/\/apisix-nginx-module/) {
 _EOC_
 }
 
+my $a6_ngx_vars = "";
```
Comment (starsz) [inline t/APISIX.pm:220]: `a6_ngx_vars` looks strange.
Maybe `apisix_ngx_vars`  would be better?

File: t/APISIX.pm (hunk 1 above)
Comment (spacewander) [reply t/APISIX.pm:222]: There is no need to append the str to `$a6_ngx_vars` which is empty if apisix-nginx-module is missing.

File: t/APISIX.pm (hunk 3 above)
Comment (spacewander) [reply t/APISIX.pm:220]: There is already a `a6_ngx_directives`, so `a6_ngx_vars` is fine

File: t/wasm/request-body.t (hunk 2 above)
Comment (spacewander) [reply t/wasm/request-body.t:152]: No. A empty `grep_error_log_out` means "should not grep any result with the pattern given in grep_error_log".

**Review by leslie-tsang — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Pull request merged**